src/sentry/api/helpers/environments.py

```diff
@@ -18,6 +20,31 @@
 }
 
 
+def get_environment(request: Request, organization_id: int) -> Environment | None:
+    environment_param = request.GET.get("environment")
+    if environment_param is None:
+        return None
+    else:
+        return Environment.get_for_organization_id(
+            name=environment_param, organization_id=organization_id
+        )
+
+
+def get_environment_id(request: Request, organization_id: int) -> int | None:
+    environment = get_environment(request, organization_id)
+    return environment.id if environment is not None else None
+
+
+def get_environment_func(
+    request: Request, organization_id: int
+) -> Callable[[], Environment | None]:
+    @functools.cache
```

Comment:
Just making sure I understand here - this cache will be unique for each invocation of `get_environment_func`?

I'm not totally sure of all the ways that this cache was useful in general. But previously, if we called `get_environment_id`, `get_environment`, etc in any order, then the cache would survive between all of them.

Would it be more helpful to store this cache in a thread.local or even in `app.env`?

Reply:
I am trying to kill `app.env` so definitely not there.  I audited all the callsites and the only time the value was reused was through `get_environment_func` -- pretty sure the cache was premature optimization anyway

Reply:
Ok, yeah I can imagine that this probably wasn't doing a lot for us since our api isn't that high volume anyway. Maybe just check out datadog after deploying and make sure reads to environment don't significantly increase?